server/src/main/java/org/opensearch/action/pagination/IndexPaginationStrategy.java

```diff
@@ -65,25 +69,47 @@ public IndexPaginationStrategy(PageParams pageParams, ClusterState clusterState)
         );
     }
 
-    private static Predicate<IndexMetadata> getMetadataFilter(String requestedTokenStr, String sortOrder) {
-        boolean isAscendingSort = sortOrder.equals(PARAM_ASC_SORT_VALUE);
-        IndexStrategyToken requestedToken = Objects.isNull(requestedTokenStr) || requestedTokenStr.isEmpty()
-            ? null
-            : new IndexStrategyToken(requestedTokenStr);
-        if (Objects.isNull(requestedToken)) {
+    protected static List<IndexMetadata> getEligibleIndices(
+        ClusterState clusterState,
+        String sortOrder,
+        String lastIndexName,
+        Long lastIndexCreationTime,
+        boolean includeLastIndex
+    ) {
+        if (Objects.isNull(lastIndexName) || Objects.isNull(lastIndexCreationTime)) {
+            return PaginationStrategy.getSortedIndexMetadata(
+                clusterState,
+                PageParams.PARAM_ASC_SORT_VALUE.equals(sortOrder) ? ASC_COMPARATOR : DESC_COMPARATOR
+            );
+        }
+        return PaginationStrategy.getSortedIndexMetadata(
+            clusterState,
+            getMetadataFilter(sortOrder, lastIndexName, lastIndexCreationTime, includeLastIndex),
+            PageParams.PARAM_ASC_SORT_VALUE.equals(sortOrder) ? ASC_COMPARATOR : DESC_COMPARATOR
+        );
+    }
+
+    protected static Predicate<IndexMetadata> getMetadataFilter(
+        String sortOrder,
+        String lastIndexName,
+        Long lastIndexCreationTime,
+        boolean includeLastIndex
```

Comment:
I think it is better to have boolean `and` [predicate](https://docs.oracle.com/javase/8/docs/api/?java/util/function/Predicate.html) 1) by lastIndexName and 2) by (lastIndexCreateTime, lastIndexName)

Reply:
Updated in the revision.